mllib/src/test/scala/org/apache/spark/ml/recommendation/ALSSuite.scala

```diff
@@ -599,8 +599,15 @@ class ALSSuite
           (ex, act) =>
             ex.userFactors.first().getSeq[Float](1) === act.userFactors.first.getSeq[Float](1)
         } { (ex, act, _) =>
-          ex.transform(_: DataFrame).select("prediction").first.getDouble(0) ~==
-            act.transform(_: DataFrame).select("prediction").first.getDouble(0) absTol 1e-6
+          testTransformerByGlobalCheckFunc[Float](_: DataFrame, ex, "prediction") {
+            case exRows: Seq[Row] =>
```

Comment:
I think it's ok to keep ex.transform here. This way the code will be a bit simpler.

Reply:
Fixed.